pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -118,84 +118,113 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
+			v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+				Type:    apiServiceAvailable,
+				Status:  operatorv1.ConditionFalse,
+				Message: strings.Join(missingAPIMessages, "\n"),
+				Reason:  "APIServiceNotAvailable",
+			})
+		} else {
+			for _, apiService := range actualAPIServices {
+				for _, condition := range apiService.Status.Conditions {
+					if condition.Type == apiregistrationv1.Available {
+						if condition.Status == apiregistrationv1.ConditionFalse {
+							availableConditionReasons = append(availableConditionReasons, "APIServiceNotAvailable")
+							availableConditionMessages = append(availableConditionMessages, fmt.Sprintf("apiservice/%v: not available: %v", apiService.Name, condition.Message))
+							v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+								Type:    apiServiceAvailable,
+								Status:  operatorv1.ConditionFalse,
+								Message: strings.Join(availableConditionMessages, "\n"),
+								Reason:  strings.Join(availableConditionReasons, "\n"),
+							})
+						}
+					}
+				}
+			}
+		}
 	}
-
 	// If the daemonset is up to date and the operatorConfig are up to date, then we are no longer progressing
 	var progressingMessages []string
 	if actualDaemonSet != nil && actualDaemonSet.ObjectMeta.Generation != actualDaemonSet.Status.ObservedGeneration {
 		progressingMessages = append(progressingMessages, fmt.Sprintf("daemonset/apiserver.openshift-operator: observed generation is %d, desired generation is %d.", actualDaemonSet.Status.ObservedGeneration, actualDaemonSet.ObjectMeta.Generation))
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeProgressing,
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "DesiredStateNotYetAchieved",
+			Message: strings.Join(progressingMessages, "\n"),
+		})
 	}
 	if operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration {
 		progressingMessages = append(progressingMessages, fmt.Sprintf("openshiftapiserveroperatorconfigs/instance: observed generation is %d, desired generation is %d.", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation))
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:    operatorv1.OperatorStatusTypeProgressing,
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "DesiredStateNotYetAchieved",
+			Message: strings.Join(progressingMessages, "\n"),
+		})
 	}
 
 	if len(progressingMessages) == 0 {
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
 			Type:   operatorv1.OperatorStatusTypeProgressing,
 			Status: operatorv1.ConditionFalse,
 		})
-	} else {
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
-			Type:    operatorv1.OperatorStatusTypeProgressing,
```

Comment:
Clean this `Progressing` condition up in addition to the `Available` one now that you are replacing with two more specific `*Progressing` conditions.

Reply:
I was wondering if I remove this condition, then how will I handle the below test cases? 
`Default`: https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00_test.go#L46
`ConfigAndDaemonSetGenerationMismatch`: https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00_test.go#L99
What could be the `conditionType` for the below test case?
`MultipleObservedAhead`: https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00_test.go#L90

Reply:
Add the new `conditionType` testcases struct and fill in the appropriate `*Progressing` type for the test case (instead of just using `OperatorStatusTypeProgressing`.

Reply:
ack